Source/Core/AST/GotoBoogie/ImplicitJump.cs

```diff
@@ -0,0 +1,10 @@
+#nullable enable
+using Microsoft.Boogie;
+
+namespace VCGeneration;
+
+public class ImplicitJump : TokenWrapper {
```

Comment:
Is it possible that this token could nest another nested token? If that is the case, I would prefer to see a parameter bool ImplicitJump
inside TokenWrapper which should then be a record, to avoid what we experienced in Dafny with unexpected piles of token wrappers that are hard to unfold properly, especially when the order should not matter.
You can then have an extension method method `WithImplicit(this IToken token)` that, if the token is already wrapped, copy it while setting the implicit flag, and otherwise wrap it with a wrapped token accordingly

Reply:
Great callout. I've changed the check from `originalReturn.tok is ImplicitJump` to `originalReturn.tok is not [Source]Token` so that any wrapping, which I think indicates it's generated code, will prevent a separate VC from being generated.

I find it hard to say what implementation is the most maintainable here, without having more cases of token wrapping to reason with, so I hope we can leave it at this and stay vigilant.